### PR TITLE
win_virtio_driver_install_by_installer: add new test case

### PR DESCRIPTION
**Pull request by nickzhq**
win_virtio_driver_install_by_installer: add new test case

add new test cases to win_virtio_driver_install_by_installer.py and win_virtio_driver_installer_repair

id: 2143143, 2143454

Signed-off-by: Houqi (Nick) Zuo hzuo@redhat.com

### CODE REVIEW

**Comment by nickzhq** [comment]
@xiagao @zhencliu @YongxueHong  Please take a look, thanks!

**Comment by YongxueHong** [comment]
Hi @nickzhq 
From your changes, I can see you add two cases.
Would you like to separate them into two commits?
Thanks.
By the way, please better control the length of the character (max 79) for each line.

**Comment by nickzhq** [comment]
> Hi @nickzhq From your changes, I can see you add two cases. Would you like to separate them into two commits? Thanks. By the way, please better control the length of the character (max 79) for each line.

Done, please check again! Thanks!

**Comment by xiagao** [comment]
I didn't know why sending "Change VM memory to 3953" twice?
[stdlog] 2023-02-02 02:31:47,540 avocado.test INFO | Context: Running envict test
[stdlog] 2023-02-02 02:31:47,540 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' 
[stdlog] 2023-02-02 02:31:47,540 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'xq9akJkt'}
[stdlog] 2023-02-02 02:31:47,540 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted)
[stdlog] 2023-02-02 02:31:47,541 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1)    actual: 4294967296
[stdlog] 2023-02-02 02:31:47,541 avocado.test INFO | Context: Running envict test --> (run_ballooning_test) --> Check memory status before ballooning test
[stdlog] 2023-02-02 02:31:47,541 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' 
[stdlog] 2023-02-02 02:31:47,541 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'aSGpj2aS'}
[stdlog] 2023-02-02 02:31:47,542 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted)
[stdlog] 2023-02-02 02:31:47,542 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1)    actual: 4294967296
[stdlog] 2023-02-02 02:31:47,542 avocado.virttest.virt_vm DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 360s)
[stdlog] 2023-02-02 02:31:47,542 avocado.virttest.virt_vm DEBUG| Found/Verified IP 10.73.75.13 for VM avocado-vt-vm1 NIC 0
[stdlog] 2023-02-02 02:31:48,324 aexpect.client DEBUG| Sending command: systeminfo
[stdlog] 2023-02-02 02:31:51,012 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' 
[stdlog] 2023-02-02 02:31:51,013 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': '2HSuIRFD'}
[stdlog] 2023-02-02 02:31:51,014 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted)
[stdlog] 2023-02-02 02:31:51,014 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1)    actual: 4294967296
[stdlog] 2023-02-02 02:31:51,014 avocado.virttest.virt_vm DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 360s)
[stdlog] 2023-02-02 02:31:51,014 avocado.virttest.virt_vm DEBUG| Found/Verified IP 10.73.75.13 for VM avocado-vt-vm1 NIC 0
[stdlog] 2023-02-02 02:31:51,807 aexpect.client DEBUG| Sending command: systeminfo
[stdlog] 2023-02-02 02:31:54,461 avocado.test INFO | Context: Running envict test --> (run_ballooning_test) --> Change VM memory to 3953
[stdlog] 2023-02-02 02:31:54,461 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'balloon' 
[stdlog] 2023-02-02 02:31:54,461 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'balloon', 'arguments': {'value': 4145020928}, 'id': 'BnD0wdSO'}
[stdlog] 2023-02-02 02:31:55,464 avocado.virttest.qemu_monitor INFO | BALLOON_CHANGE QMP event received
[stdlog] 2023-02-02 02:31:55,464 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' 
[stdlog] 2023-02-02 02:31:55,464 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'pVIoaSBR'}
[stdlog] 2023-02-02 02:31:55,465 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted)
[stdlog] 2023-02-02 02:31:55,465 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1)    actual: 4145020928
[stdlog] 2023-02-02 02:31:55,465 avocado.test INFO | Context: Running envict test --> (run_ballooning_test) --> Check memory status after envict memory
[stdlog] 2023-02-02 02:31:55,465 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' 
[stdlog] 2023-02-02 02:31:55,465 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'gC8zj5UD'}
[stdlog] 2023-02-02 02:31:55,466 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted)
[stdlog] 2023-02-02 02:31:55,466 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1)    actual: 4145020928
[stdlog] 2023-02-02 02:31:55,466 avocado.virttest.virt_vm DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 360s)
[stdlog] 2023-02-02 02:31:55,466 avocado.virttest.virt_vm DEBUG| Found/Verified IP 10.73.75.13 for VM avocado-vt-vm1 NIC 0
[stdlog] 2023-02-02 02:31:56,272 aexpect.client DEBUG| Sending command: systeminfo
[stdlog] 2023-02-02 02:31:59,954 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' 
[stdlog] 2023-02-02 02:31:59,955 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'YHLWfsY7'}
[stdlog] 2023-02-02 02:31:59,956 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted)
[stdlog] 2023-02-02 02:31:59,956 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1)    actual: 4145020928
[stdlog] 2023-02-02 02:31:59,956 avocado.virttest.virt_vm DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 360s)
[stdlog] 2023-02-02 02:31:59,956 avocado.virttest.virt_vm DEBUG| Found/Verified IP 10.73.75.13 for VM avocado-vt-vm1 NIC 0
[stdlog] 2023-02-02 02:32:00,743 aexpect.client DEBUG| Sending command: systeminfo
[stdlog] 2023-02-02 02:32:03,402 avocado.test INFO | Context: Running envict test --> (run_ballooning_test) --> Change VM memory to 3953
[stdlog] 2023-02-02 02:32:03,403 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'balloon' 
[stdlog] 2023-02-02 02:32:03,403 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'balloon', 'arguments': {'value': 4145020928}, 'id': 'iV6wztDX'}
[stdlog] 2023-02-02 02:34:03,527 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' 
[stdlog] 2023-02-02 02:34:03,527 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'F3YXvjQS'}
[stdlog] 2023-02-02 02:34:03,528 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted)
[stdlog] 2023-02-02 02:34:03,529 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1)    actual: 4145020928
[stdlog] 2023-02-02 02:34:03,529 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' 
[stdlog] 2023-02-02 02:34:03,529 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'qsqZw2Ma'}
[stdlog] 2023-02-02 02:34:03,529 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted)
[stdlog] 2023-02-02 02:34:03,530 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1)    actual: 4145020928
[stdlog] 2023-02-02 02:34:03,530 avocado.test INFO | Context: Running envict test --> (run_ballooning_test) --> Check memory status after envict memory
[stdlog] 2023-02-02 02:34:03,530 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' 
[stdlog] 2023-02-02 02:34:03,530 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'Hq61dGz6'}
[stdlog] 2023-02-02 02:34:03,530 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted)
[stdlog] 2023-02-02 02:34:03,530 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1)    actual: 4145020928
[stdlog] 2023-02-02 02:34:03,531 avocado.virttest.virt_vm DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 360s)
[stdlog] 2023-02-02 02:34:03,531 avocado.virttest.virt_vm DEBUG| Found/Verified IP 10.73.75.13 for VM avocado-vt-vm1 NIC 0
[stdlog] 2023-02-02 02:34:04,327 aexpect.client DEBUG| Sending command: systeminfo
[stdlog] 2023-02-02 02:34:07,172 avocado.test ERROR| Memory size mismatch after envict memory:
[stdlog] 2023-02-02 02:34:07,173 avocado.test ERROR| Wanted to be changed: 0.0
[stdlog] Changed in monitor: 0.0
[stdlog] Changed in guest: 340

**Comment by nickzhq** [comment]
> I didn't know why sending "Change VM memory to 3953" twice?
> 
> [stdlog] 2023-02-02 02:31:54,461 avocado.test INFO | Context: Running envict test --> (run_ballooning_test) --> Change VM memory to 3953 [stdlog] 2023-02-02 02:31:54,461 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'balloon' [stdlog] 2023-02-02 02:31:54,461 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'balloon', 'arguments': {'value': 4145020928}, 'id': 'BnD0wdSO'} [stdlog] 2023-02-02 02:31:55,464 avocado.virttest.qemu_monitor INFO | BALLOON_CHANGE QMP event received [stdlog] 2023-02-02 02:31:55,464 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' [stdlog] 2023-02-02 02:31:55,464 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'pVIoaSBR'} [stdlog] 2023-02-02 02:31:55,465 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted) [stdlog] 2023-02-02 02:31:55,465 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) actual: 4145020928 [stdlog] 2023-02-02 02:31:55,465 avocado.test INFO | Context: Running envict test --> (run_ballooning_test) --> Check memory status after envict memory [stdlog] 2023-02-02 02:31:55,465 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' [stdlog] 2023-02-02 02:31:55,465 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'gC8zj5UD'} [stdlog] 2023-02-02 02:31:55,466 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted) [stdlog] 2023-02-02 02:31:55,466 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) actual: 4145020928 [stdlog] 2023-02-02 02:31:55,466 avocado.virttest.virt_vm DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 360s) [stdlog] 2023-02-02 02:31:55,466 avocado.virttest.virt_vm DEBUG| Found/Verified IP 10.73.75.13 for VM avocado-vt-vm1 NIC 0 [stdlog] 2023-02-02 02:31:56,272 aexpect.client DEBUG| Sending command: systeminfo [stdlog] 2023-02-02 02:31:59,954 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' [stdlog] 2023-02-02 02:31:59,955 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'YHLWfsY7'} [stdlog] 2023-02-02 02:31:59,956 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted) [stdlog] 2023-02-02 02:31:59,956 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) actual: 4145020928 [stdlog] 2023-02-02 02:31:59,956 avocado.virttest.virt_vm DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 360s) [stdlog] 2023-02-02 02:31:59,956 avocado.virttest.virt_vm DEBUG| Found/Verified IP 10.73.75.13 for VM avocado-vt-vm1 NIC 0 [stdlog] 2023-02-02 02:32:00,743 aexpect.client DEBUG| Sending command: systeminfo [stdlog] 2023-02-02 02:32:03,402 avocado.test INFO | Context: Running envict test --> (run_ballooning_test) --> Change VM memory to 3953 [stdlog] 2023-02-02 02:32:03,403 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'balloon' [stdlog] 2023-02-02 02:32:03,403 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'balloon', 'arguments': {'value': 4145020928}, 'id': 'iV6wztDX'} [stdlog] 2023-02-02 02:34:03,527 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' [stdlog] 2023-02-02 02:34:03,527 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'F3YXvjQS'} [stdlog] 2023-02-02 02:34:03,528 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted) [stdlog] 2023-02-02 02:34:03,529 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) actual: 4145020928 [stdlog] 2023-02-02 02:34:03,529 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' [stdlog] 2023-02-02 02:34:03,529 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'qsqZw2Ma'} [stdlog] 2023-02-02 02:34:03,529 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted) [stdlog] 2023-02-02 02:34:03,530 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) actual: 4145020928 [stdlog] 2023-02-02 02:34:03,530 avocado.test INFO | Context: Running envict test --> (run_ballooning_test) --> Check memory status after envict memory [stdlog] 2023-02-02 02:34:03,530 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Sending command 'query-balloon' [stdlog] 2023-02-02 02:34:03,530 avocado.virttest.qemu_monitor DEBUG| Send command: {'execute': 'query-balloon', 'id': 'Hq61dGz6'} [stdlog] 2023-02-02 02:34:03,530 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) Response to 'query-balloon' (re-formatted) [stdlog] 2023-02-02 02:34:03,530 avocado.virttest.qemu_monitor DEBUG| (monitor avocado-vt-vm1.qmp1) actual: 4145020928 [stdlog] 2023-02-02 02:34:03,531 avocado.virttest.virt_vm DEBUG| Attempting to log into 'avocado-vt-vm1' (timeout 360s) [stdlog] 2023-02-02 02:34:03,531 avocado.virttest.virt_vm DEBUG| Found/Verified IP 10.73.75.13 for VM avocado-vt-vm1 NIC 0 [stdlog] 2023-02-02 02:34:04,327 aexpect.client DEBUG| Sending command: systeminfo [stdlog] 2023-02-02 02:34:07,172 avocado.test ERROR| Memory size mismatch after envict memory: [stdlog] 2023-02-02 02:34:07,173 avocado.test ERROR| Wanted to be changed: 0.0 [stdlog] Changed in monitor: 0.0 [stdlog] Changed in guest: 340

Done, please test again. Thanks!

**Comment by nickzhq** [comment]
@xiagao @zhencliu @YongxueHong  Please take a look, thanks!

**Review by xiagao — DISMISSED** [review]

**Comment by nickzhq** [comment]
@YongxueHong @zhencliu Please take a look again, thanks!

**Comment by YongxueHong** [comment]
Hi @nickzhq 
I think we have a gap in the concept tp-qemu test case.
For the tp-qemu test, we define `variants: - xxx` as a test case.
Such as:
```
variants:
     - case1:
     - case2:
```
 The above config has two cases: `case1` and `case2`.
So if we want to allocate a commit to a case, it should be related to the section of `variants:` and the corresponding changes in the py file.
You could refer to https://github.com/autotest/tp-qemu/pull/2009/commits/d3e7da730e5de221496e28817077b2c782798c11
Thanks.

**Comment by nickzhq** [comment]
@YongxueHong Please check again, thanks!

File: qemu/tests/win_virtio_driver_installer_repair.py [hunk 1]
```diff
@@ -99,6 +99,11 @@ def run(test, params, env):
             driver_test_params = params.get('driver_test_params_%s'
                                             % driver_name, '{}')
             driver_test_params = ast.literal_eval(driver_test_params)
+            if driver_name == "viofs":
+                win_driver_installer_test.run_viofs_service(test, params,
+                                                            session)
+            if driver_name != "balloon":
+                driver_test_params = ast.literal_eval(driver_test_params)
```
Comment (zhencliu) [inline qemu/tests/win_virtio_driver_installer_repair.py:107]: hi @nickzhq , I am curious about why we run ast.literal_eval twice when driver_name is viofs, do we hit error as driver_test_params_viofs is not set? Did I miss anything?

Comment (nickzhq) [reply qemu/tests/win_virtio_driver_installer_repair.py:107]: > hi @nickzhq , I am curious about why we run ast.literal_eval twice when driver_name is viofs, do we hit error as driver_test_params_viofs is not set? Did I miss anything?

Hi @zhencliu , the reason is code conflict during "git rebase", there is no worry at here. However, this mistake has been solved in third commit at this patch.

**Pull request merged**